### ISSUE TITLE
Question on .sum 

### ISSUE THREAD

**Issue by dltlr** [realm/realm-cocoa#3392]
Can I use the .sum on realm results to sum values embedded lists?

Object1 -> has a list of Object2, Object2 has a number value X

I have a result list of Object1 and want the sum of x.  results.sum("object2.x") does not work. Is this possible now or should this be a feature request?

**Comment by jpsim** [comment]
Unfortunately this is not currently supported, but we're now tracking adding support for that functionality in this ticket, so you'll be notified of any progress we make on this front.

In the meantime, you could get all the X values and sum them outside Realm:

```
import RealmSwift

class Object1: Object {
    let object2s = List<Object2>()
}

class Object2: Object {
    dynamic var x = 0
}

let realm = try! Realm()

try! realm.write {
    for i in 0..<5 {
        realm.create(Object2.self, value: [i])
    }
    realm.create(Object1.self, value: [realm.objects(Object2)])
}

let xValues = realm.objects(Object1).valueForKeyPath("object2s.x") as! [[Int]]
let xSum = xValues.flatMap({ $0 }).reduce(0, combine: +)
xSum // => 10
```

**Comment by dltlr** [comment]
Thanks!!!

**Comment by kishikawakatsumi** [comment]
@dltlr Yes. `Int64` comforms to `AddableType`. See https://github.com/realm/realm-cocoa/blob/master/RealmSwift-swift2.0/Results.swift#L45
